integrations/integrations.go

```diff
@@ -0,0 +1,232 @@
+package integrations
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	cometdb "github.com/cometbft/cometbft-db"
+	dbm "github.com/cosmos/cosmos-db"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/server"
+	serverconfig "github.com/cosmos/cosmos-sdk/server/config"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+	"github.com/polymerdao/monomer"
+	"github.com/polymerdao/monomer/environment"
+	"github.com/polymerdao/monomer/genesis"
+	"github.com/polymerdao/monomer/node"
+	"github.com/spf13/viper"
+)
+
+func StartCommandHandler(
+	svrCtx *server.Context,
+	clientCtx client.Context, //nolint:gocritic // hugeParam
+	appCreator servertypes.AppCreator,
+	inProcessConsesus bool,
+	opts server.StartCmdOptions,
+) error {
+	// We assume `inProcessConsensus` is true for now, so let's return an error if it's not.
+	if !inProcessConsesus {
+		return fmt.Errorf("in-process consensus must be enabled")
+	}
+
+	env := environment.New()
+	defer func() {
+		if closeErr := env.Close(); closeErr != nil {
+			svrCtx.Logger.Error(closeErr.Error())
+		}
+	}()
+
+	// --- Do the normal Cosmos SDK stuff ---
+	svrCfg, err := getAndValidateConfig(svrCtx)
+	if err != nil {
+		return fmt.Errorf("failed to get and validate server config: %w", err)
+	}
+
+	app, err := startApp(env, svrCtx, appCreator, opts)
+	if err != nil {
+		return fmt.Errorf("failed to start application: %w", err)
+	}
+
+	// Would usually start a Comet node in-process here, but we replace the
+	// Comet node with a Monomer node.
+	if err := startInProcess(env, svrCtx, svrCfg, clientCtx, app, opts); err != nil {
+		svrCtx.Logger.Error("failed to start Monomer node in-process", "error", err)
+		return fmt.Errorf("failed to start Monomer node in-process: %w", err)
+	}
+
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+	<-sigCh
+
+	return nil
+}
+
+var _ io.WriteCloser = (*fakeTraceWriter)(nil)
+
+type fakeTraceWriter struct{}
+
+func (f *fakeTraceWriter) Write([]byte) (int, error) {
+	return 0, nil
+}
+
+func (f *fakeTraceWriter) Close() error {
+	return nil
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L624
+func startApp(
+	env *environment.Env,
+	svrCtx *server.Context,
+	appCreator servertypes.AppCreator,
+	opts server.StartCmdOptions,
+) (app servertypes.Application, err error) {
+	home := svrCtx.Config.RootDir
+	db, err := opts.DBOpener(home, server.GetAppDBBackend(svrCtx.Viper))
+	if err != nil {
+		return app, err
```

Comment:
almost always preferred to embellish error strings. Makes it easier to track bugs down
```suggestion
		return app, fmt.Errrof("open db: %v", err")
```